migrate_fields_config.py

```diff
@@ -0,0 +1,222 @@
+    [{'v1ConfigName': 'fieldQueries', 'v2ConfigName': 'includeInQuery', 'v1Value': True, 'v2Value': False}, {'v1ConfigName': 'sort', 'v2ConfigName': 'sort', 'v1Value': False, 'v2Value': True}]
+    """
+    field_diff = [{KEY_V1_CONFIGNAME: diff,
+                   KEY_V2_CONFIGNAME: FIELDS_KEYS_V1_V2[diff],
+                   KEY_V1_VALUE: v1_field[diff],
+                   KEY_V2_VALUE: v2_field[FIELDS_KEYS_V1_V2[diff]]}
+                  for diff in FIELDS_KEYS_V1_V2
+                    if diff in v1_field and FIELDS_KEYS_V1_V2[diff] in v2_field and v1_field[diff] != v2_field[FIELDS_KEYS_V1_V2[diff]]]
+    return field_diff
+
+
+def get_fields_differences(v1_fields, v2_fields):
+    """ Get a list of differences between 2 fields list
+    >>> get_fields_differences({}, {})
+    []
+    >>> v1_fields = {'field0': {'sort': True, 'facet': False}, 'field1': {'sort': True, 'facet': True}}
+    >>> v2_fields = {'field1': {'sort': False, 'facet': False}, 'field2': {'sort': True, 'facet': True}}
+    >>> get_fields_differences(v1_fields, v2_fields)
+    [('field1', {'v1Field': {'sort': True, 'facet': True}, 'v2Field': {'sort': False, 'facet': False}, 'configDiff': [{'v1ConfigName': 'facet', 'v2ConfigName': 'facet', 'v1Value': True, 'v2Value': False}, {'v1ConfigName': 'sort', 'v2ConfigName': 'sort', 'v1Value': True, 'v2Value': False}]})]
+    """
+    v1_field_names = v1_fields.keys()
+    v2_field_names = v2_fields.keys()
+    diffs = list()
+    for v1_field_name in v1_field_names:
+        if v1_field_name in v2_field_names:
+            v1_field = v1_fields[v1_field_name]
+            v2_field = v2_fields[v1_field_name]
+            diff = get_fields_difference(v1_field, v2_fields[v1_field_name])
+            if diff:
+                diffs.append((v1_field_name, {KEY_V1_FIELD: v1_field, KEY_V2_FIELD: v2_field, KEY_CONFIG_DIFF: diff}))
+    return diffs
+
+
+def v2_get_updated_field(field_difference):
+    """ Get a modified field according to the differences provided (CloudV2)
+    >>> diffs = ('field1', {'v1Field': {'sort': True, 'facet': True}, 'v2Field': {'name': 'field1', 'sort': False, 'facet': False, 'type': 'SOMETYPE'}, 'configDiff': [{'v1ConfigName': 'facet', 'v2ConfigName': 'facet', 'v1Value': True, 'v2Value': False}, {'v1ConfigName': 'sort', 'v2ConfigName': 'sort', 'v1Value': True, 'v2Value': False}]})
+    >>> v2_get_updated_field(diffs)
+    {'name': 'field1', 'sort': True, 'facet': True, 'type': 'SOMETYPE'}
+    """
+    for diff in field_difference[1][KEY_CONFIG_DIFF]:
+        # print(f'\t- "{diff[KEY_V2_CONFIGNAME]}" is "{diff[KEY_V2_VALUE]}" but should be "{diff[KEY_V1_VALUE]}"')
+        if field_difference[1][KEY_V2_FIELD]['type'].lower() == 'date' and \
+                diff[KEY_V2_CONFIGNAME].lower() == 'sort' and \
+                diff[KEY_V1_VALUE] == False:
+            print(f'\t-> Field "{field_difference[1][KEY_V2_FIELD]["name"]}" is of type date and cannot be set to sort = false. Skipping this change.')
```

Comment:
Would that really happen?

A V2 field that is of type date has to be sort = true is that it?

Reply:
Yes, I saw it in V1. In V2 however, sort needs to be to True if it's a date field.